lib/iris/fileformats/grib/message.py

```diff
@@ -258,10 +253,9 @@ def __getitem__(self, keys):
                 # Only the non-masked values are included in codedValues.
                 _data = np.empty(shape=bitmap.shape)
                 _data[bitmap.astype(bool)] = data
-                # Use nan where input = 1, the opposite of the behaviour
-                # specified by the GRIB spec.
-                _data[np.logical_not(bitmap.astype(bool))] = np.nan
-                data = _data
+                # `ma.masked_array` masks where input = 1, the opposite of
+                # the behaviour specified by the GRIB spec.
+                data = ma.masked_array(_data, mask=np.logical_not(bitmap))
```

Comment:
@lbdreyer So the deal is that `_DataProxy` will return a data payload that is masked, right? Rather than an array that is nan filled ... that might have testing fall-out, I guess.

That aside, shouldn't this be ...
```python
data = ma.masked_array(_data, mask=np.logical_not(bitmap.astype(bool)), fill_value=np.nan)
```
This would align with the strategy for auto-selecting the `fill_value` on save, right?

... Only problem is that using `fill_value=np.nan` means that the `dtype` of all GRIB data must be `float`, otherwise it's not possible to set `np.nan` as a `fill_value` for non-`float` data ...